src/testlib.ts

```diff
@@ -1,29 +1,80 @@
-const EXECUTE_RESPONSE_SCHEMA = require('../intents/execute.response.schema.json');
-const DISCONNECT_RESPONSE_SCHEMA = require('../intents/disconnect.response.schema.json');
-const ON_OFF_STATES_SCHEMA = require('../traits/onoff.states.schema.json');
-const ON_OFF_ATTRIBUTES_SCHEMA = require('../traits/onoff.attributes.schema.json');
-
+/**
+ * Retrieve the JSON schema for the given intent.
+ * @param intentName FQN of the intent
+ * @param schemaType One of 'request' or 'response'
+ */
+function getIntentSchema(intentName: string, schemaType: string): object | undefined {
+  try {
+    const shortIntent = intentName.split('.').pop()?.toLowerCase() || 'unknown';
+    const filename = `${shortIntent}.${schemaType}.schema.json`;
+    return require(path.join('../schema/intents', shortIntent, filename));
+  } catch (error) {
+    console.error(error);
+    return undefined;
+  }
+}
 
-const QUERY_COMMAND_STATES_EXPECT = {
-  'action.devices.traits.OnOff': ON_OFF_STATES_SCHEMA,
-};
+/**
+ * Retrieve the JSON schema for the given device trait.
+ * @param traitName FQN of the device trait
+ * @param schemaType One of 'attributes' or 'states'
+ */
+function getTraitSchema(traitName: string, schemaType: string): object | undefined {
+  try {
+    const shortTrait = traitName.split('.').pop()?.toLowerCase() || 'unknown';
+    const filename = `${shortTrait}.${schemaType}.schema.json`;
+    return require(path.join('../schema/traits', shortTrait, filename));
+  } catch (error) {
+    console.error(traitName, error);
+    return undefined;
+  }
+}
 
-const EXECUTE_COMMAND_STATES_EXPECT = {
-  'action.devices.commands.OnOff': ON_OFF_STATES_SCHEMA,
-};
+/**
+ * Retrieve JSON schema for the given device command
+ * @param commandName FQN of the device command
+ * @param schemaType One of 'params' or 'results'
+ */
+function getCommandSchema(commandName: string, schemaType: string): object| undefined {
+  try {
+    const shortTrait = findEnclosingTrait(commandName, schemaType) || 'unknown';
+    const shortCommand = commandName.split('.').pop()?.toLowerCase() || 'unknown';
+    const filename = `${shortCommand}.${schemaType}.schema.json`;
+    const basePath = '../schema/traits';
+    return require(path.join(basePath, shortTrait, filename));
+  } catch (error) {
+    console.error(error);
+    return undefined;
+  }
+}
 
-const TRAIT_ATTRIBUTES_EXPECT = {
-  'action.devices.traits.OnOff': ON_OFF_ATTRIBUTES_SCHEMA,
-};
+/**
+ * Helper function to search for matching trait directory
+ * @param commandName
+ * @param schemaType
+ */
+function findEnclosingTrait(commandName: string, schemaType: string) {
+  try {
+    const shortCommand = commandName.split('.').pop()?.toLowerCase() || 'unknown';
+    const filename = `${shortCommand}.${schemaType}.schema.json`;
+    const basePath = path.join(__dirname, '../schema/traits');
+    return fs.readdirSync(basePath).find((entry) =>
+      fs.readdirSync(path.join(basePath, entry)).includes(filename)
```

Comment:
rather than reading the whole dir you could just check that the file exist.

Reply:
Good catch. Done.